devito/ir/clusters/analysis.py

```diff
@@ -131,11 +131,15 @@ def _callback(self, clusters, d, prefix):
                 is_parallel_indep &= (dep.distance_mapper.get(d.root) == 0)
                 continue
 
-            if not dep.is_increment:
-                return SEQUENTIAL
+            if dep.is_increment:
+                if dep.function in scope.initialized:
+                    # False alarm, the increment is over a locally-defined symbol
+                    pass
+                else:
+                    is_parallel_atomic = True
```

Comment:
`is_parallel_atomic = is_parallel_atomic or True` or only last `dep` required.

Reply:
? 

no, I think the current way is correct: as soon as the first `dep` inducing an increment is encountered, then we are inside a parallel-atomic dimension.

IOW, you may have 1000 deps, and 999 of them could be fully parallel, but if only one of them requires an atomic increment, then the whole dimension is parallel-atomic

Reply:
it clearly doesn't matter whether it's the first or last dep

Reply:
I could actually add an optimization that we don't waste time computing `test00` and `test01` and `test1` if we know already that `is_parallel_atomic is True`, because at that point the only worse thing that can happen is that the dimension turns out to be `SEQUENTIAL`